src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,488 @@
+                _terminal.WriteError(StringUtil.Loc("AutoLogonPoliciesWarningsHeader"));
+                for (int i=0; i < warningReasons.Count; i++)
+                {
+                    var msg = String.Format("{0} - {1}", i + 1, warningReasons[i]);
+                    Trace.Warning(msg);
+                    _terminal.WriteError(msg);
+                }
+                _terminal.WriteLine();
+            }
+        }
+
+        private void UpdateUserSpecificRegistrySettings(CommandSettings command, string securityId)
+        {
+            //User specific
+            UpdateScreenSaverSettings(command, securityId);
+            
+            //User specific
+            string subKeyName = $"{securityId}\\{RegistryConstants.UserSettings.SubKeys.StartupProcess}";
+            TakeBackupAndSetValue(RegistryHive.Users, subKeyName, RegistryConstants.UserSettings.ValueNames.StartupProcess, GetStartupCommand());
+        }
+
+        private void UpdateScreenSaverSettings(CommandSettings command, string securityId)
+        {
+            if(!command.GetDisableScreenSaver())
+            {
+                return;
+            }
+
+            Trace.Info("Checking for policies that may prevent screensaver from being disabled.");
+            _terminal.WriteLine(StringUtil.Loc("ScreenSaverPoliciesInspection"));
+
+            string subKeyName = $"{securityId}\\{RegistryConstants.UserSettings.SubKeys.ScreenSaverDomainPolicy}";
+            if(_registryManager.SubKeyExists(RegistryHive.Users, subKeyName))
+            {            
+                var screenSaverValue = _registryManager.GetValue(RegistryHive.Users, subKeyName, RegistryConstants.UserSettings.ValueNames.ScreenSaver);
+                if (int.TryParse(screenSaverValue, out int isScreenSaverDomainPolicySet)
+                        && isScreenSaverDomainPolicySet == 1)
+                {
+                    Trace.Warning("Screensaver policy is defined on the machine. Screensaver may not remain disabled always.");
+                    _terminal.WriteError(StringUtil.Loc("ScreenSaverPolicyWarning"));
+                }
+            }
+
+            string screenSaverSubKeyName = $"{securityId}\\{RegistryConstants.UserSettings.SubKeys.ScreenSaver}";
+            TakeBackupAndSetValue(RegistryHive.Users, screenSaverSubKeyName, RegistryConstants.UserSettings.ValueNames.ScreenSaver, "0");
+        }
+
+        private string GetStartupCommand()
+        {
+            //startup process            
+            string cmdExePath = Environment.GetEnvironmentVariable("comspec");
+            if (string.IsNullOrEmpty(cmdExePath))
+            {
+                Trace.Error("Unable to get the path for cmd.exe.");
+                throw new Exception(StringUtil.Loc("FilePathNotFound", "cmd.exe"));
+            }
+
+            //file to run in cmd.exe
+            var filePath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Root), "run.cmd");
```

Comment:
should you call agent.listener instead of run.cmd?
i think run.cmd should send `--startuptype interactive` as default